### PR TITLE
Expand `NamedExpr` range to include full range of parenthesized value

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Given:

```python
if (
    x
    :=
    (  # 4
        y # 5
    )  # 6
):
    pass
```

It turns out the parser ended the range of the `NamedExpr` at the end of `y`, rather than the end of the parenthesis that encloses `y`. This just seems like a bug -- the range should be from the start of the name on the left, to the end of the parenthesized node on the right.

## Test Plan

`cargo test`

### CODE REVIEW

File: crates/ruff_python_parser/src/python.lalrpop [hunk 1]
```diff
@@ -1335,15 +1335,19 @@ NamedExpressionTest: ast::Expr = {
...
+
 NamedExpression: ast::Expr = {
-    <location:@L> <id:Identifier> <end_location:@R> ":=" <value:Test<"all">> => {
+    <location:@L> <target:NamedExpressionName> ":=" <value:Test<"all">> <end_location:@R> => {
         ast::Expr::NamedExpr(
             ast::ExprNamedExpr {
-                target: Box::new(ast::Expr::Name(
-                    ast::ExprName { id: id.into(), ctx: ast::ExprContext::Store, range: (location..end_location).into() },
-                )),
-                range: (location..value.end()).into(),
```
Comment (charliermarsh) [inline crates/ruff_python_parser/src/python.lalrpop:1345]: In the example from the summary, `value.end()` is the end of `y`, not the end of its parenthesized range.

File: crates/ruff_python_parser/src/python.rs
Comment (MichaReiser) [inline crates/ruff_python_parser/src/python.rs:1]: Why change 18k lines???

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_python_parser/src/python.rs
Comment (charliermarsh) [reply crates/ruff_python_parser/src/python.rs:1]: Oh maybe I have the wrong version again? Or maybe I committed at some point with the wrong version?

Comment (charliermarsh) [reply crates/ruff_python_parser/src/python.rs:1]: I re-ran locally on latest but still see huge changes, we must've committed an old version in the past or something.

**Pull request merged**